splash/lua_modules/wraputils.lua

```diff
@@ -109,6 +109,31 @@ local function sets_callback(func, storage)
   end
 end
 
+local function pack_callback_return_value(func)
+  return function(cb, ...)
+    local mcb = cb
+
+    if type(cb) == 'function' then
+      mcb = function(...)
+        return { cb(...) }
+      end
+    end
+
+    return func(mcb, ...)
+  end
+end
```

Comment:
I think we can find a smarter way of doing it automatically, but don't have anything specific to back my claim. Let's keep it for now then.